### DISCUSSION TITLE
Understanding DEFAULT_ROOT

### DISCUSSION THREAD

**Discussion by evertramos** [nginx-proxy/nginx-proxy#1924] [Q&A]
Hello @buchdag and community!

I am using version 1.0.0 and tried it out the DEFAULT ROOT option, but could not really succed. Might misunderstand the usage of this option. Here is what I thought:

### nginx-proxy option for default response

First time I read, I thoutht this option would create a default root to respond from my nginx when no virtual host was found, I know we have DEFAULT_HOST, but in that case (DEFAULT_ROOT) we would use a redirect route to external sites, and no need to have an internal container to catch all the requested urls if there is no VIRTUAL_HOST configured.

So, this setup did not work for me... here is what I tried:

1. My nginx-proxy container
```console
docker run -d \
     -p 80:80 \
     -v /var/run/docker.sock:/tmp/docker.sock:ro \
     -e DEFAULT_ROOT=418 \
     --net proxy \
     --name nginx-proxy \
     nginxproxy/nginx-proxy
```

2. nginx-proxy container inspection
 ```json
           "Env": [
                "SSL_POLICY=Mozilla-Intermediate",
                "DEFAULT_ROOT=418",
                "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
                "NGINX_VERSION=1.21.6",
                "NJS_VERSION=0.7.2",
                "PKG_RELEASE=1~bullseye",
                "NGINX_PROXY_VERSION=1.0.0-4-g4ea3437",
                "DOCKER_GEN_VERSION=0.8.2",
                "DOCKER_HOST=unix:///tmp/docker.sock"
            ],
            "Cmd": [
                "forego",
                "start",
                "-r"
            ],
            "Image": "nginxproxy/nginx-proxy",
```

3. when trying in the browser a domain poiting out to my server got nginx 503 default response

### VIRTUAL_HOST option

Then I read again and though that this option (DEFAULT_ROOT) would work along the VIRTUAL_PATH and VIRTUAL_DEST, which worke awesomely, thank you very much for this new features!

But I still could not make it work, the response for a unknown path was still 404:

1. My container
```json
            "Env": [
                "VIRTUAL_HOST=foo.ezyhost.com.br",
                "VIRTUAL_PATH=/gone/",
                "VIRTUAL_DEST=/",
                "DEFAULT_ROOT=418",
                "PATH=/usr/local/apache2/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
                "HTTPD_PREFIX=/usr/local/apache2",
                "HTTPD_VERSION=2.4.46",
                "HTTPD_SHA256=740eddf6e1c641992b22359cabc66e6325868c3c5e2e3f98faf349b61ecf41ea",
                "HTTPD_PATCHES="
            ],
            "Cmd": [
                "httpd-foreground"
            ],
            "Image": "httpd:alpine",
            "Volumes": {
                "/usr/local/apache2/htdocs": {}
            },
```

2. The nginx-proxy result
```nginx
   # foo.ezyhost.com.br/gone/
  upstream foo.ezyhost.com.br-2d8dc8dc58afb8ce2d30c03f7960db00a4ab5638 {
        ## Can be connected with "proxy" network
        # subsite
        server 172.24.0.3:80;
  }
server {
  server_name foo.ezyhost.com.br;
  listen 80 ;
  access_log /var/log/nginx/access.log vhost;
  include /etc/nginx/vhost.d/default;
  location /gone/ {
    proxy_pass http://foo.ezyhost.com.br-2d8dc8dc58afb8ce2d30c03f7960db00a4ab5638/;
  }
  location / {
    return 404;
  }
}
```

So, what am I missing?

**Comment by buchdag** [comment]
Hi Evert. The `DEFAULT_ROOT` feature is indeed meant to be used with the path based routing, [but it's not working per container, you're supposed to set it globally on the nginx-proxy container](https://github.com/nginx-proxy/nginx-proxy/blob/4ea3437dfab423b6710b32b2aa9d50f9f750bed9/nginx.tmpl#L8). 

It'll set the root response when you use the path based routing without a container set to the root path for a given `VIRTUAL_HOST` set.
